File: LeetCode/Java Solutions/Element Appearing More Than 25% In Sorted Array/element.java
class Solution {
    public int findSpecialInteger(int[] arr) {
        int n = arr.length;
        int threshold = n / 4;

        int currentCount = 1;
        for (int i = 1; i < n; i++) {
            if (arr[i] == arr[i - 1]) {
                currentCount++;
                if (currentCount > threshold) {
                    return arr[i];
                }
            } else {
                currentCount = 1;
            }
        }

        return arr[0]; // Handle case for single element array
    }
}

File: LeetCode/Python Solutions/Element Appearing More Than 25% In Sorted Array/element.py
class Solution:
    def findSpecialInteger(self, arr: List[int]) -> int:
        n = len(arr)
        threshold = n // 4  # 25% of the length of the array

        current_count = 1
        for i in range(1, n):
            if arr[i] == arr[i - 1]:
                current_count += 1
                if current_count > threshold:
                    return arr[i]
            else:
                current_count = 1

        return arr[0]  # Handle case for single element array

